hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/CommitWatcher.java

```diff
@@ -84,6 +84,13 @@ public CommitWatcher(BufferPool bufferPool, XceiverClientSpi xceiverClient) {
     futureMap = new ConcurrentHashMap<>();
   }
 
+  public CommitWatcher(XceiverClientSpi xceiverClient) {
```

Comment:
Let's copy CommitWatcher to a new class, say StreamCommitWatcher?  In StreamCommitWatcher, we don't need bufferPool.  The commitIndex2flushedDataMap should be different -- it should not use ChunkBuffer anymore.